test/ysh-runtime-errors.sh

```diff
@@ -421,6 +421,10 @@ test-dict-construct() {
   _expr-error-case '= Dict([["too", "many", "parts"]])'
 }
 
+test-bound-func() {
+  _expr-error-case '= "foo"->startswith(1)' # shouldn't panic
```

Comment:
can we add more of the errors in this file?  It's useful to be able to eyeball them all at one

Like `test-precise-error-locations() {`